docs/src/architecture/secureboot.md

```diff
@@ -0,0 +1,50 @@
+<!--
+    Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+    SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# Secure Boot
+
+This section describes secure boot and how to create secure keys.
+
+## Process of Enabling
+
+Secure Boot can be enabled on NixOS using the project Lanzaboote. Secure Boot is a UEFI feature that only allows trusted operating systems to boot.
+Lanzaboote has two components: lzbt and stub. lzbt signs and installs the boot files on the ESP. stub is a UEFI application that loads the kernel and initrd from the ESP.
+
+### Process of Secure keys creation
+
+To create Secure Boot keys, we use sbctl, a popular Secure Boot Key Manager. sbctl is available in Nixpkgs as pkgs.sbctl.
+
+Once you have installed sbctl (or entered a Nix shell), creating your Secure Boot keys requires this command:
+```
+$ sudo sbctl create-keys
+```
+
+Using above method users can create secure keys they trust
+
+### Current Implementation
+
+For demonstration purposes we are using pre-generated secure keys which is UNSECURE as whosoever has keys can break into the system.
```

Comment:
`For demonstration purposes, we use pre-generated secure keys which are **unsecure** as whoever has keys can break into the system.`